src/main/java/pl/wrapper/parking/infrastructure/inMemory/InMemoryRepositoryImpl.java

```diff
@@ -0,0 +1,83 @@
+package pl.wrapper.parking.infrastructure.inMemory;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
+import java.io.*;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import org.springframework.core.serializer.support.SerializationFailedException;
+import org.springframework.scheduling.annotation.Scheduled;
+
+public abstract class InMemoryRepositoryImpl<K extends Serializable, V extends Serializable>
+        implements InMemoryRepository<K, V> {
+
+    protected final transient File file;
+    protected Map<K, V> dataMap;
+    protected V defaultValue;
+
+    public InMemoryRepositoryImpl(String filePath, Map<K, V> map, V defaultValue) {
+        this.file = new File(filePath);
+        this.defaultValue = defaultValue;
+
+        this.dataMap = map;
+    }
+
+    @Override
+    public void add(K key, V value) {
+        if (key == null || value == null)
```

Comment:
We don't really need it. I mean, we can trust people using this method not to do that, right...?